src/rust/dns/lib.rs

```diff
@@ -0,0 +1,174 @@
+#[cxx::bridge(namespace = "workerd::rust::dns")]
+mod ffi {
+    /// CAA record representation
+    struct CaaRecord {
+        critical: u8,
+        field: String,
+        value: String,
+    }
+    /// NAPTR record representation
+    struct NaptrRecord {
+        flags: String,
+        service: String,
+        regexp: String,
+        replacement: String,
+        order: u32,
+        preference: u32,
+    }
+    extern "Rust" {
+        fn parse_caa_record(record: &str) -> CaaRecord;
+        fn parse_naptr_record(record: &str) -> NaptrRecord;
+    }
+}
+
+/// Given a vector of strings, converts each slice to UTF-8 from HEX.
+///
+/// # Panics
```

Comment:
We shouldn't panic. Especially during parsing. Let's replace this with errors. I suggest using `thiserror` - very easy to use.